packages/api/src/graphql/mutation.public.ts

```diff
@@ -536,6 +538,49 @@ export const GraphQLPublicMutation = new GraphQLObjectType<undefined, Context>({
           failureURL
         })
       }
+    },
+
+    checkInvoiceStatus: {
+      type: GraphQLPublicInvoice,
+      args: {
+        id: {type: GraphQLNonNull(GraphQLID)}
+      },
+      description:
+        'This mutation will check the invoice status and update with information from the paymentProvider',
+      async resolve(root, {id}, context) {
+        const {authenticateUser, dbAdapter, paymentProviders} = context
+        const {user} = authenticateUser()
+
+        const invoices = await dbAdapter.invoice.getInvoicesByUserID(user.id)
+        const invoice = invoices.find(invoice => invoice !== null && invoice.id === id)
+        if (!invoice) throw new NotFound('Invoice', id)
+
+        const payments = await dbAdapter.payment.getPaymentsByInvoiceID(invoice.id)
+        const paymentMethods = await dbAdapter.paymentMethod.getActivePaymentMethods()
+
+        for (const payment of payments) {
+          if (!payment || !payment.intentID) continue
```

Comment:
Maybe:
```
if (!payment?.intentID) continue
```

Reply:
it's not the same here. Here i'm checking for undefined